src/libmongoc/src/mongoc/mongoc-structured-log.c

```diff
@@ -269,32 +271,54 @@ mongoc_structured_log_get_named_component (const char *name, mongoc_structured_l
    return false;
 }
 
-static int32_t
-_mongoc_structured_log_get_max_document_length_from_env (void)
+size_t
+mongoc_structured_log_opts_get_max_document_length (const mongoc_structured_log_opts_t *opts)
 {
+   return (size_t) opts->max_document_length;
+}
+
+bool
+mongoc_structured_log_opts_set_max_document_length (mongoc_structured_log_opts_t *opts, size_t max_document_length)
+{
+   if (max_document_length <= (size_t) MONGOC_STRUCTURED_LOG_MAXIMUM_MAX_DOCUMENT_LENGTH) {
+      opts->max_document_length = (uint32_t) max_document_length;
+      return true;
+   } else {
+      return false;
+   }
+}
+
+bool
+mongoc_structured_log_opts_set_max_document_length_from_env (mongoc_structured_log_opts_t *opts)
+{
+   BSON_ASSERT_PARAM (opts);
+
    const char *variable = "MONGODB_LOG_MAX_DOCUMENT_LENGTH";
    const char *max_length_str = getenv (variable);
 
    if (!max_length_str) {
-      return MONGOC_STRUCTURED_LOG_DEFAULT_MAX_DOCUMENT_LENGTH;
+      return true;
    }
 
    if (!strcasecmp (max_length_str, "unlimited")) {
-      return BSON_MAX_LEN_UNLIMITED;
+      BSON_ASSERT (
+         mongoc_structured_log_opts_set_max_document_length (opts, MONGOC_STRUCTURED_LOG_MAXIMUM_MAX_DOCUMENT_LENGTH));
+      return true;
    }
 
    char *endptr;
    long int_value = strtol (max_length_str, &endptr, 10);
-   if (int_value >= 0 && int_value <= INT32_MAX && endptr != max_length_str && !*endptr) {
-      return (int32_t) int_value;
+   if (int_value >= 0 && endptr != max_length_str && !*endptr && mcommon_in_range_signed (size_t, int_value) &&
```

Comment:
```suggestion
   if (int_value >= 0 && endptr != max_length_str && *endptr == '\0' && mcommon_in_range_signed (size_t, int_value) &&
```
Recommend against treating characters as booleans.